safer_remote_clusters.md

```diff
@@ -0,0 +1,43 @@
+# Safer Remote Clusters (proposal)
+
+### Problem
+
+Users often change their kube context, forget about it, `tilt up`, and end up
+deploying to prod.
+
+### Solutions
+
+The following are not mutually exclusive!
+
+#### Point users at a pure Starlark Solution
+
+a la [what dmiller came up with](https://gist.github.com/jazzdan/d30ef73482920beb3ac9fa2fcb2c3146)
+
+Pros:
+* cheap
+* flexible
+
+Cons:
+* not very discoverable
+* unsafe by default
+
+#### Require intervention when deploying over a non-Tilt-deployed entity
+
+Pros:
+* would be nice to have regardless of anything else we do
+
+Cons:
+* not clear we have a good existing UI pattern for such an intervention
+* doesn't protect users from deploying non-prod resources to prod (e.g., I've `tilt up`'d both integration tests and servantes in prod this week), though that's obviously a much less dire problem than killing production
+
+#### Require remote k8s clusters to be explicitly whitelisted
+
+Add an `allowed_remote_k8s_clusters` Tiltfile directive. Generate a Tiltfile error unless either:
+1. the k8s context api's host is localhost
+2. the k8s context's name is passed to `allowed_remote_k8s_clusters`
+
+Pros:
+* pretty simple UI
+
+Cons:
+* gets kind of weird in the staging case - sometimes I want Tilt to be able to deploy there, and sometimes I don't. I'd have to edit the Tiltfile every time. This might be unavoidable.
```

Comment:
maybe this is additionally configurable by flags? (flags being acceptable here b/c it's never a value we want to change mid-tilt run)

e.g. your tiltfile only whitelists D4D, but when you explicitly want to deploy to staging, you set your kubeconfig to point to your staging cluster and `tilt up --whitelist-cluster=123.456.789` or whatever

(probably no need to implement this out of the gate, but seems a good solution to this con if ppl complain)